File: Java/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> participantMap = new HashMap();
        
        for (String str: participant) {
            participantMap.put(str, participantMap.getOrDefault(str, 0) + 1);
        }
        
        for (String str: completion) {
            participantMap.put(str, participantMap.get(str) - 1);
        }
        
        String answer = "";
        for (String str: participantMap.keySet()) {
            if (participantMap.get(str) > 0) {
                answer = str;
            }
        }
        
        return answer;
    }
}

File: Python3/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    d = {}
    for x in participant:
        d[x] = d.get(x, 0) + 1
    
    for x in completion:
        d[x] -= 1
    
    dnf = [k for k, v in d.items() if v > 0]
    
    return dnf[0]




""" 풀이2
from collections import Counter

def solution(participant, completion):
    return ''.join(Counter(participant) - Counter(completion))
"""
    